<?xml version="1.0" encoding="utf-8" ?>
<!--
SPDX-License-Identifier: BSD-3-Clause
SPDX-FileCopyrightText: Czech Technical University in Prague
-->
<launch>
    <arg name="camera" default="camera" />

    <node pkg="robot_model_renderer" type="robot_model_renderer" name="camera_self_filter" output="screen">
        <!-- input topic for camera_info -->
        <remap from="camera_info" to="$(arg camera)/camera_info" />
        <!-- ouput topic for mask -->
        <remap from="mask" to="$(arg camera)/mask" />

        <!-- Default value of parameter "invert_colors=false" means that pixels where is robot have white color(255) and
             other pixels black color(0). If value of parameter is true then pixels have opposite values."   -->
        <param name="invert_colors" value="false" />
        <param name="near_clip" value="0.03" />
        <!-- Outline around the mask. -->
        <param name="outline_width" value="5" />
        <!-- If true, apply camera distortion to the rendered image, otherwise render rectified image. -->
        <param name="do_distort" value="true" />
    </node>
</launch>
